pkg/controller/common/helpers.go

```diff
@@ -22,3 +25,19 @@ func WriteTerminationError(err error) {
 	ioutil.WriteFile("/dev/termination-log", []byte(msg), 0644)
 	glog.Fatal(msg)
 }
+
+// ValidateIgnition evaluates whether a given non-empty Ignition Config is valid
+// Returns nil if the Ignition Config is empty or if valid
+// Returns an error message containing a Report otherwise
+func ValidateIgnition(cfg igntypes.Config) error {
+	emptyIgnition := igntypes.Config{}
+
+	// only validate if Ignition Config is not empty
+	if !reflect.DeepEqual(emptyIgnition, cfg) {
+		report := validate.ValidateWithoutSource(reflect.ValueOf(cfg))
+		if report.IsFatal() {
+			return errors.Errorf("invalid Ignition config found: %v", report)
+		}
+	}
+	return nil
```

Comment:
logic is perfect, I'd just invert the `if` to only have an indent level and less vars (readability mostly), but not blocking:

```go
        if reflect.DeepEqual(igntypes.Config{}, cfg) {
            return nil
        }
	if report := validate.ValidateWithoutSource(reflect.ValueOf(cfg)); report.IsFatal() {
		return errors.Errorf("invalid Ignition config found: %v", report)
	}
	return nil
```

Reply:
ahh i had it this way at first but switched it.  will make this change thanks!

Reply:
Done.